consumer.go

```diff
@@ -479,23 +479,27 @@ func (child *partitionConsumer) parseResponse(response *FetchResponse) ([]*Consu
 	prelude := true
 	var messages []*ConsumerMessage
 	for _, msgBlock := range block.MsgSet.Messages {
-
 		for _, msg := range msgBlock.Messages() {
-			if prelude && msg.Offset < child.offset {
+			offset := msg.Offset
+			if child.conf.Version.IsAtLeast(V0_10_0_0) {
+				// calculate correct offset from relative offset
+				offset = msgBlock.Offset - int64(len(msgBlock.Messages())) + msg.Offset + 1
```

Comment:
Per the comment in the java version (https://github.com/apache/kafka/blob/e49b3aee6b25b9b26275849c693d5dec41d6cf68/clients/src/main/java/org/apache/kafka/common/record/MemoryRecords.java#L248-L249) and a careful reading of the KIP that @edenhill linked, I don't know if we can just assume that the offset of the final message is equal to the number of messages in the set? It sounds like the broker is supposed to re-assign discontiguous compressed relative offsets, but the java client handles them being discontiguous anyway?

Tangentially, having the base outer offset be the offset of the *last* message instead of the *first* message feels really bizarre. That's definitely the way it works, but I can't for the life of me figure out why.